cmake/BLTPrivateMacros.cmake

```diff
@@ -433,6 +459,99 @@ macro(blt_setup_cuda_target)
     endif()
 endmacro(blt_setup_cuda_target)
 
+##------------------------------------------------------------------------------
+## blt_patch_target( NAME <targetname>
+##                     DEPENDS_ON [dep1 [dep2 ...]]
+##                     INCLUDES [include1 [include2 ...]]
+##                     LIBRARIES [lib1 [lib2 ...]]
+##                     TREAT_INCLUDES_AS_SYSTEM [ON|OFF]
+##                     FORTRAN_MODULES [ path1 [ path2 ..]]
+##                     COMPILE_FLAGS [ flag1 [ flag2 ..]]
+##                     LINK_FLAGS [ flag1 [ flag2 ..]]
+##                     DEFINES [def1 [def2 ...]] )
+##
+## Modifies an existing CMake target - currently only sets INTERFACE visibility
+##------------------------------------------------------------------------------
+macro(blt_patch_target)
+    set(singleValueArgs NAME TREAT_INCLUDES_AS_SYSTEM)
+    set(multiValueArgs INCLUDES 
+                       DEPENDS_ON
+                       LIBRARIES
+                       FORTRAN_MODULES
+                       COMPILE_FLAGS
+                       LINK_FLAGS
+                       DEFINES )
+
+    ## parse the arguments
+    cmake_parse_arguments(arg
+        "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    # Input checks
+    if( "${arg_NAME}" STREQUAL "" )
+        message(FATAL_ERROR "blt_patch_target() must be called with argument NAME <name>")
+    endif()
+
+    if (NOT TARGET ${arg_NAME})
+        message(FATAL_ERROR "blt_patch_target() NAME argument must be a native CMake target")
+    endif()
+
+    # LIBRARIES and DEPENDS_ON are kept separate in case different logic is needed for
+    # the library itself versus its dependencies
+    if( arg_LIBRARIES )
+        target_link_libraries(${arg_NAME} INTERFACE ${arg_LIBRARIES})
+    endif()
+
+    # TODO: This won't expand BLT-registered libraries
+    if( arg_DEPENDS_ON )
+        target_link_libraries(${arg_NAME} INTERFACE ${arg_DEPENDS_ON})
+    endif()
+
+    if( arg_INCLUDES )
+        if( ${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.11.0" )
+            target_include_directories(${arg_NAME} INTERFACE ${arg_INCLUDES})
+            # PGI does not support -isystem
+            if( (${arg_TREAT_INCLUDES_AS_SYSTEM}) AND (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI"))
```

Comment:
This use-case would be very helpful:

```
blt_patch_target(NAME axom TREAT_INCLUDES_AS_SYSTEM ON)
```

Where it would just copy the includes to the system includes variable.  We do this a lot in various projects. https://github.com/LLNL/axom/blob/f609592e34c5ce08facebc3c393e3aec68617ea9/src/cmake/thirdparty/SetupAxomThirdParty.cmake#L93

Reply:
So in addition to exposing `blt_patch_target` as a public macro, you'd prefer to have TREAT_INCLUDES_AS_SYSTEM apply to all include directories and not just the ones specified by the INCLUDES parameter?

Reply:
Private vs public: I would say yes, I think this is a very helpful macro that should be exposed and documented.

SYSTEM_INCLUDE: Yes, all customers I have talked to have wanted a target's includes to stop producing warnings want all of them to.

Reply:
I agree re the system include stuff.